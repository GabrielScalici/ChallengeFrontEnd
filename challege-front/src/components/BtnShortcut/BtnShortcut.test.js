import React from 'react';
import { shallow } from 'enzyme';
import ConnectedApp, { BtnShortcut } from './index';
import checkPropTypes from 'check-prop-types';
import renderer from 'react-test-renderer';


const setUp = (props = {}) => {
    const component = shallow(<BtnShortcut {...props} />);
    return component;
}

describe('Component show shortcut button render', () => {

    let component;
    beforeEach(() => {
        component = setUp();
    });

    describe('render with show = false', () => {

        let wrapper;
        beforeEach(() => {
            const props = {
                price: '20,00'
            };
            wrapper = setUp(props);
        });

        //COMPONENTS
        it('render only button', () => {
            const component = wrapper.find('.BtnShortcut');
            expect(component.length).toBe(1);
        });

        it('render price', () => {
            const component = wrapper.find('.BtnShortcutPrice');
            expect(component.length).toBe(0);
        });

        it('render with mouse event', () => {
            const tree = renderer.create(<BtnShortcut/>).toJSON();

            expect(tree).toMatchSnapshot();
        });


    });


    describe('Checking PropTypes', () => {

        const simple_func = jest.fn();

        it('Verify props type string and function', () => {

            const expectedProps = {
                title: "Teste",
                onClick: simple_func,
            };

            //PROP TYPE
            const propsErr = checkPropTypes(BtnShortcut.propTypes, expectedProps, 'props', BtnShortcut.name);
            expect(propsErr).toBeUndefined();
        });


    });

});