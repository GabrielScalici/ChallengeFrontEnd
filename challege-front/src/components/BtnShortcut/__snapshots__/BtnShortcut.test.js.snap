// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component show shortcut button render Link changes the class when hovered 1`] = `
<div
  className="BtnShortcut"
>
  <button
    className="ui massive blue button"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</div>
`;

exports[`Component show shortcut button render render with show = false render with mouse event 1`] = `
<div
  className="BtnShortcut"
>
  <button
    className="ui massive blue button"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</div>
`;
